File: room.java
import java.util.ArrayList;

public class Room extends Location {
 
 private String name;
 private String desc;
 private ArrayList<Item> contents;
 private int numberOfItems;
 public Room() {
 contents = new ArrayList<>();
 setType("Room");
    }
 
 public Room(String name, String desc) {
 this.name = name;
 this.desc = desc;
 contents = new ArrayList<>();
 setType("Room");
    }

 public ArrayList<Items>  getContents() {
 return contents;
    }
 
 public void addObject(Item item){
 contents.add(item);
    }
 public void dropObject(Item item){
 contents.remove(item);
    }
 

 public void setContents(ArrayList<Item> contents) {
 this.contents = contents;
    }

 public int getNumberOfItems() {
 numberOfItems = contents.size();
 return numberOfItems;
    }

 public String getName() {
 return name;
    }
 public void setName(String name) {
 this.name = name;
    }
 public String getDesc() {
 return desc;
    }
 public void setDesc(String desc) {
 this.desc = desc;
    }
 public String setLocationType() {
 return "Room";
 
    }
 
 public void viewRoomContent() {
 System.out.println("Room Contents");
 System.out.println();
 if(contents.size()==0)
 System.out.println("No Item in Room");
 else {
 for(int i = 0 ; i < contents.size(); i++)
 System.out.println(i + "  " + contents.get(i));
        }
 System.out.println();
    }

    @Override
 public String toString() {
 return "Room [name=" + name + ", desc=" + desc + ", contents=" + contents
                + ", numberOfItemss=" + getNumberOfItemss() + "]";
    }

 public Item getRoomItem(int takeItem) {
 Item t = contents.get(takeItem);
 dropObject(t);
 return t;
 
    }
}

File: Room.js
class Room extends Location {
    constructor(name, desc) {
      super();
      this.name = name;
      this.desc = desc;
      this.contents = [];
      this.numberOfItems = 0;
      this.setType("Room");
    }
  
    getContents() {
      return this.contents;
    }
  
    addObject(item) {
      this.contents.push(item);
    }
  
    dropObject(item) {
      const index = this.contents.indexOf(item);
      if (index > -1) {
        this.contents.splice(index, 1);
      }
    }
  
    setContents(contents) {
      this.contents = contents;
    }
  
    getNumberOfItems() {
      this.numberOfItems = this.contents.length;
      return this.numberOfItems;
    }
  
    getName() {
      return this.name;
    }
  
    setName(name) {
      this.name = name;
    }
  
    getDesc() {
      return this.desc;
    }
  
    setDesc(desc) {
      this.desc = desc;
    }
  
    setLocationType() {
      return "Room";
    }
  
    viewRoomContent() {
      console.log("Room Contents");
      console.log();
      if (this.contents.length === 0) {
        console.log("No Item in Room");
      } else {
        for (let i = 0; i < this.contents.length; i++) {
          console.log(i + "  " + this.contents[i]);
        }
      }
      console.log();
    }
  
    toString() {
      return "Room [name=" + this.name + ", desc=" + this.desc + ", contents=" + this.contents + ", numberOfItems=" + this.getNumberOfItems() + "]";
    }
  
    getRoomItem(takeItem) {
      const item = this.contents[takeItem];
      this.dropObject(item);
      return item;
    }
 
    


}
  
  